src/main/java/seedu/address/logic/commands/order/DoneCommand.java

```diff
@@ -53,12 +54,19 @@ public CommandResult execute(Model model, CommandHistory history) throws Command
         }
 
         orderToBeCompleted.setStatusCompleted();
+
+        //fetch deliveryman from index because order's deliveryman not reliable.
         Deliveryman deliverymanToRemoveOrder = orderToBeCompleted.getDeliveryman();
-        Deliveryman updatedDeliveryman = removeOrderFromDeliveryman(deliverymanToRemoveOrder, orderToBeCompleted);
+        int indexOfDeliveryman = lastShowDeliverymanList.indexOf(deliverymanToRemoveOrder);
+        Deliveryman correctDeliveryman = lastShowDeliverymanList.get(indexOfDeliveryman);
+        Deliveryman updatedDeliveryman = removeOrderFromDeliveryman(correctDeliveryman, orderToBeCompleted);
```

Comment:
will it be cleaner if we do something like 
```
correctDeliveryman = lastShowDeliverymanList.stream()
        .filter(dman -> dman.equals(deliverymanToRemoveOrder))
        .findFirst
        .orElseThrow(…some appropriate exception)
```
This allows for some level of defensiveness as well (ensures that the deliveryman exists, even though it should be the case).
(this is inspired by the `OrderListPanelHandle` in tests.)